diff --git a/docroot/sites/all/modules/invite/modules/invite_by_email/invite_by_email.module b/docroot/sites/all/modules/invite/modules/invite_by_email/invite_by_email.module
index bbc95b9..a7a4fe1 100644
--- a/docroot/sites/all/modules/invite/modules/invite_by_email/invite_by_email.module
+++ b/docroot/sites/all/modules/invite/modules/invite_by_email/invite_by_email.module
@@ -344,7 +344,9 @@ function invite_by_email_send_invitation($invite) {
   $existing_invite = invite_load($invite->reg_code);
   if ($existing_invite) {
     $invite->expiry = REQUEST_TIME + (variable_get('invite_default_expiry_time', 30) * 60 * 60 * 24);
-    $invite->resent++;
+    if (isset($invite->resent)) {
+      $invite->resent++;
+    }
   }
 
   $entity = entity_metadata_wrapper('invite', $invite);
